ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+        MMInitBoundingBox(&pCurrentPolHeader->dfBB);
+
+        pCurrentPolHeader->dfPerimeter = 0;
+        pCurrentPolHeader->dfArea = 0L;
+    }
+
+    // Setting flushes to all sections described in
+    // format specifications document.
+    pFlushAL = &pMMArc->FlushAL;
+    pFlushNL = &pMMNode->FlushNL;
+    pFlushZL = &pMMArc->pZSection.FlushZL;
+    pFlushPS = &hMiraMonLayer->MMPolygon.FlushPS;
+    pFlushPAL = &hMiraMonLayer->MMPolygon.FlushPAL;
+
+    pFlushNL->pBlockWhereToSaveOrRead = (void *)pMMNode->pNL;
+    pFlushAL->pBlockWhereToSaveOrRead = (void *)pMMArc->pAL;
+    if (hMiraMonLayer->TopHeader.bIs3d)
+        pFlushZL->pBlockWhereToSaveOrRead = (void *)pMMArc->pZSection.pZL;
+    if (hMiraMonLayer->bIsPolygon)
+    {
+        pFlushPS->pBlockWhereToSaveOrRead =
+            (void *)hMiraMonLayer->MMPolygon.pPS;
+        pFlushPAL->pBlockWhereToSaveOrRead =
+            (void *)hMiraMonLayer->MMPolygon.pPAL;
+    }
+
+    // Creation of the MiraMon extended database
+    if (!hMiraMonLayer->bIsPolygon)
+    {
+        if (hMiraMonLayer->TopHeader.nElemCount == 0)
+        {
+            MMCPLDebug("MiraMon", "Creating MiraMon database");
+            if (MMCreateMMDB(hMiraMonLayer))
+                return MM_FATAL_ERROR_WRITING_FEATURES;
+            MMCPLDebug("MiraMon", "MiraMon database created");
+        }
+    }
+    else
+    {  // Universal polygon has been created
+        if (hMiraMonLayer->TopHeader.nElemCount == 1)
+        {
+            MMCPLDebug("MiraMon", "Creating MiraMon database");
+            if (MMCreateMMDB(hMiraMonLayer))
+                return MM_FATAL_ERROR_WRITING_FEATURES;
+            MMCPLDebug("MiraMon", "MiraMon database created");
+
+            // Universal polygon have a record with ID_GRAFIC=0 and blancs
+            if (MMAddPolygonRecordToMMDB(hMiraMonLayer, nullptr, 0, 0, nullptr))
+                return MM_FATAL_ERROR_WRITING_FEATURES;
+        }
+    }
+
+    // Checking if its possible continue writing the file due
+    // to version limitations.
+    if (hMiraMonLayer->LayerVersion == MM_32BITS_VERSION)
+    {
+        MM_FILE_OFFSET nNodeOffset, nArcOffset;
+        MM_INTERNAL_FID nArcElemCount, nNodeElemCount;
+        nNodeOffset = pFlushNL->TotalSavedBytes + pFlushNL->nNumBytes;
+        nArcOffset = pMMArc->nOffsetArc;
+
+        nArcElemCount = pArcTopHeader->nElemCount;
+        nNodeElemCount = pNodeTopHeader->nElemCount;
+        for (nIPart = 0; nIPart < hMMFeature->nNRings; nIPart++,
+            nArcElemCount++,
+            nNodeElemCount += (hMiraMonLayer->bIsPolygon ? 1 : 2))
+        {
+            // There is space for the element that is going to be written?
+            // Polygon or arc
+            if (MMCheckVersionForFID(hMiraMonLayer,
+                                     hMiraMonLayer->TopHeader.nElemCount))
+            {
+                MMCPLDebug("MiraMon", "Error in MMCheckVersionForFID() (1)");
```

Comment:
those debug messages are actually error messages, so MMCPLError() should be used. Same for below similar instances

Reply:
Changed by MMCPLError(CE_Failure, CPLE_NotSupported, "Error in MMCheckVersionForFID() (1)");